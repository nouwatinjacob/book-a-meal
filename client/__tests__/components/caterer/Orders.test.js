import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  Orders,
  mapStateToProps
} from '../../../src/components/caterer/Orders.jsx';
import {
  catererOrderResponse
} from '../../__mockData__/mockOrder';

const props = {
  getAllCatererOrderAction: () => Promise.resolve(),
  orderState: {
    orderDetails: catererOrderResponse,
    paginate: catererOrderResponse.paginate
  }
};

describe('Orders component', () => {
  const setup = () => shallow(<Orders {...props} />);

  it('should render Orders component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });
});

describe('Orders component', () => {
  const propss = {
    getAllCatererOrderAction: () => Promise.resolve(),
    orderState: {
      orderDetails: catererOrderResponse.orders,
      paginate: catererOrderResponse.paginate
    }
  };

  const setup = () => shallow(<Orders {...propss} />);

  it(`should call 'handlePageClick'
  if pagination button is clicked`, (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePageClickSpy = jest.spyOn(
      wrapper.instance(),
      'handlePageClick'
    );
    wrapper.instance().handlePageClick(data);
    expect(handlePageClickSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handleDateChange'
  when date is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        value: '2018-05-09'
      }
    };

    const handleDateChangeSpy = jest.spyOn(
      wrapper.instance(),
      'handleDateChange'
    );
    wrapper.instance().handleDateChange(event);
    expect(handleDateChangeSpy).toHaveBeenCalled();

    done();
  });
});
