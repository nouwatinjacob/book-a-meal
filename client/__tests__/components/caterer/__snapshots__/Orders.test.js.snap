// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders component should render Orders component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <div
    className="wrapper"
  >
    <div
      className="row"
    >
      <div
        className="c-medium-6 c-small-12"
        id="pd-0"
      >
        <h5>
          Sort By Date:
          <input
            onChange={[Function]}
            style={
              Object {
                "width": "150px",
              }
            }
            type="date"
          />
        </h5>
      </div>
      <div
        className="c-medium-6 c-small-12 revenue"
        id="pd-0"
      >
        <h5>
          Today's Revenue: 
          <strong>
            â‚¦
            0
          </strong>
        </h5>
      </div>
    </div>
  </div>
  <div
    className="wrapper"
  >
    <h3
      className="text-center"
    >
      No Order Found for this date
    </h3>
  </div>
  <br />
  <br />
</div>
`;
